<?xml version="1.0" encoding="utf-8"?>
<launch>

  <!-- this file is compatible with different robot models -->

  <!-- following are the arguments you can pass this launch file -->
  <!-- gazebo environment -->
  <arg name="paused" default="false"/>
  <arg name="debug" default="false"/>
  <arg name="use_sim_time" default="true"/>
  <arg name="gui" default="true"/>
  <arg name="headless" default="false"/>
  <!-- experimental setup of the robots -->
  <arg name="robot_name" default="morph_bot"/>

  <!-- we resume the logic in empty_world.launch, changing the name of the world to be launched -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find morph_bot_description)/worlds/empty.world"/>
    <arg name="paused" value="$(arg paused)"/>
    <arg name="debug" value="$(arg debug)" />
    <arg name="gui" value="$(arg gui)" />
    <arg name="use_sim_time" value="$(arg use_sim_time)"/>
    <arg name="headless" value="$(arg headless)"/>
  </include>

  <!-- parameter server -->
  <!-- all parameters under morph_sim namespace -->
  <!-- robot_name decide which robot is used in the simulation -->
  <param name="/morph_sim/robot_name" type="str" value="$(arg robot_name)"/>
  <!-- list all the robot model files here, use the one on robot_name parameter for swarm robot simulation -->
  <param name="/morph_sim/two_wheel_robot_urdf" type="str" textfile="$(find morph_bot_description)/urdf/two_wheel_robot.urdf"/>



</launch>
